src/common/notifications_winrt/BackgroundHandler.cpp

```diff
@@ -0,0 +1,26 @@
+﻿#include "pch.h"
+#include "BackgroundHandler.h"
+#include "BackgroundHandler.g.cpp"
+
+#include "handler_functions.h"
+
+namespace winrt::PowerToysNotifications::implementation
+{
+    using Windows::ApplicationModel::Background::IBackgroundTaskInstance;
+    using Windows::UI::Notifications::ToastNotificationActionTriggerDetail;
+
+    void BackgroundHandler::Run(IBackgroundTaskInstance bti)
+    {
+        const auto details = bti.TriggerDetails().try_as<ToastNotificationActionTriggerDetail>();
+        if (!details)
+        {
+            return;
+        }
+        std::wstring button_id_and_handler_data{ details.Argument() };
+        std::wstring_view button_id_and_handler{ button_id_and_handler_data };
+        const auto first_semicolon_pos = button_id_and_handler.find(';');
```

Comment:
Not a big fan of this type of parsing, can we make it a proper XML and parse it with an XML parser?

Reply:
Toast notifications docs use [URI encoding](https://docs.microsoft.com/en-us/uwp/api/windows.foundation.uri.queryparsed#Windows_Foundation_Uri_QueryParsed) extensively, so we also could probably use it.